src/overlays/actors/ovl_En_Daiku/z_en_daiku.c

```diff
@@ -23,39 +70,552 @@ const ActorInit En_Daiku_InitVars = {
+    Actor_MoveForward(&this->actor);
+    func_8002E4B4(globalCtx, &this->actor, 0.0f, 0.0f, 0.0f, 4);
+
+    if (this->subCamActive) {
+        EnDaiku_UpdateSubCamera(this, globalCtx);
+        if (this->escapeSubCamTimer-- <= 0) {
+            EnDaiku_EscapeSuccess(this, globalCtx);
+        }
+    }
+
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+}
+
+void EnDaiku_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnDaiku* this = THIS;
+    s32 curFrame;
+    Player* player = PLAYER;
+
+    if (this->currentAnimIndex == ENDAIKU_ANIM_RUN) {
+        curFrame = this->skelAnime.animCurrentFrame;
+        if (curFrame == 6 || curFrame == 15) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_MORIBLIN_WALK);
+        }
+    }
+
+    Collider_CylinderUpdate(&this->actor, &this->collider);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Daiku/EnDaiku_Destroy.s")
+    this->actionFunc(this, globalCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Daiku/func_809E2F50.s")
+    if (this->stateFlags & ENDAIKU_STATEFLAG_1) {
+        this->unk_244.unk_18.x = player->actor.posRot2.pos.x;
+        this->unk_244.unk_18.y = player->actor.posRot2.pos.y;
+        this->unk_244.unk_18.z = player->actor.posRot2.pos.z;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Daiku/func_809E3068.s")
+        if (this->stateFlags & ENDAIKU_STATEFLAG_2) {
+            func_80034A14(&this->actor, &this->unk_244, 0, 4);
+        } else {
+            func_80034A14(&this->actor, &this->unk_244, 0, 2);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Daiku/func_809E3338.s")
+void EnDaiku_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnDaiku* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Daiku/func_809E336C.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_daiku.c", 1227);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Daiku/func_809E3438.s")
+    func_80093D18(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Daiku/func_809E349C.s")
+    if ((thisx->params & 3) == 0) {
+        gDPSetEnvColor(POLY_OPA_DISP++, 170, 10, 70, 255);
+    } else if ((thisx->params & 3) == 1) {
+        gDPSetEnvColor(POLY_OPA_DISP++, 170, 200, 255, 255);
+    } else if ((thisx->params & 3) == 2) {
+        gDPSetEnvColor(POLY_OPA_DISP++, 0, 230, 70, 255);
+    } else if ((thisx->params & 3) == 3) {
+        gDPSetEnvColor(POLY_OPA_DISP++, 200, 0, 150, 255);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Daiku/func_809E36A4.s")
+    SkelAnime_DrawFlexOpa(globalCtx, this->skelAnime.skeleton, this->skelAnime.limbDrawTbl, this->skelAnime.dListCount,
+                          EnDaiku_OverrideLimbDraw, EnDaiku_PostLimbDraw, &this->actor);
```

Comment:
I believe we just pass `this` now
```suggestion
                          EnDaiku_OverrideLimbDraw, EnDaiku_PostLimbDraw, this);
```